src/test/xmeta/test_symbol_table.cpp

```diff
@@ -110,3 +110,176 @@ TEST_CASE("Enum circular dependency")
+    REQUIRE(properties[2]->get_set_method()->get_id() == "set_property_3");
+    REQUIRE(!properties[2]->is_get_method_declared_first());
+    REQUIRE(properties[3]->get_get_method() != nullptr);
+    REQUIRE(properties[3]->get_set_method() != nullptr);
+    REQUIRE(properties[3]->get_get_method()->get_id() == "get_property_4");
+    REQUIRE(properties[3]->get_set_method()->get_id() == "set_property_4");
+    REQUIRE(properties[3]->is_get_method_declared_first());
+    REQUIRE(properties[4]->get_get_method() != nullptr);
+    REQUIRE(properties[4]->get_set_method() != nullptr);
+    REQUIRE(properties[4]->get_get_method()->get_id() == "get_property_5");
+    REQUIRE(properties[4]->get_set_method()->get_id() == "set_property_5");
+    REQUIRE(!properties[4]->is_get_method_declared_first());
+
+    auto classes = ns_body->get_classes();
+    REQUIRE(classes.size() == 1);
+    REQUIRE(classes.find("C") != classes.end());
+    auto c = classes.at("C");
+    properties = c->get_properties();
+    REQUIRE(properties.size() == 5);
+    REQUIRE(properties[0]->get_id() == "property_1");
+    REQUIRE(properties[1]->get_id() == "property_2");
+    REQUIRE(properties[2]->get_id() == "property_3");
+    REQUIRE(properties[3]->get_id() == "property_4");
+    REQUIRE(properties[4]->get_id() == "property_5");
+    REQUIRE(properties[0]->get_get_method() != nullptr);
+    REQUIRE(properties[0]->get_get_method()->get_id() == "get_property_1");
+    REQUIRE(properties[0]->get_set_method() != nullptr);
+    REQUIRE(properties[0]->get_set_method()->get_id() == "set_property_1");
+    REQUIRE(properties[1]->get_get_method() != nullptr);
+    REQUIRE(properties[1]->get_set_method() != nullptr);
+    REQUIRE(properties[1]->get_get_method()->get_id() == "get_property_2");
+    REQUIRE(properties[1]->get_set_method()->get_id() == "set_property_2");
+    REQUIRE(properties[1]->is_get_method_declared_first());
+    REQUIRE(properties[2]->get_get_method() != nullptr);
+    REQUIRE(properties[2]->get_set_method() != nullptr);
+    REQUIRE(properties[2]->get_get_method()->get_id() == "get_property_3");
+    REQUIRE(properties[2]->get_set_method()->get_id() == "set_property_3");
+    REQUIRE(!properties[2]->is_get_method_declared_first());
+    REQUIRE(properties[3]->get_get_method() != nullptr);
+    REQUIRE(properties[3]->get_set_method() != nullptr);
+    REQUIRE(properties[3]->get_get_method()->get_id() == "get_property_4");
+    REQUIRE(properties[3]->get_set_method()->get_id() == "set_property_4");
+    REQUIRE(properties[3]->is_get_method_declared_first());
+    REQUIRE(properties[4]->get_get_method() != nullptr);
+    REQUIRE(properties[4]->get_set_method() != nullptr);
+    REQUIRE(properties[4]->get_get_method()->get_id() == "get_property_5");
+    REQUIRE(properties[4]->get_set_method()->get_id() == "set_property_5");
+    REQUIRE(!properties[4]->is_get_method_declared_first());
+
+    reader.reset("");
+    REQUIRE(reader.read(incorrect_property_test_idl) == 0);
```

Comment:
What is this return value for? Seems like an implementation detail that should be abstracted away so we don't have to perform as much surgery down the road.

Reply:
read() is returning the number of syntax error that the parser encountered. This makes sense in a test but you're right, for xmeta.exe, it might be more useful to return errors with more information instead.